docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch4.tla:55:78}}
+```
+
+Let's start with these two definitions:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch4.tla:60:61}}
+```
+
+As you have probably guessed, we define two (local) values `mid` and `midVal`.
+The value `mid` is the average of `low` and `high`. The operator `\div` is
+simply integer division, which is usually written as `/` in programming
+languages. The value `midVal` is the value at the location `mid + 1`. Since
+the TLA+ sequence `INPUT_SEQ` has indices in the range `1..Len(INPUT_SEQ)`,
+whereas we are computing zero-based indices, we are adjusting the index by one,
+that is, we write `INPUT_SEQ[mid + 1]` instead of `INPUT_SEQ[mid]`.
+
+*Warning: The definitions of `mid` and `midVal` do not properly reflect the
+Java code of `binarySearch`. We will fix them later. It is a good exercise
+to stop here and think about the source of this imprecision.*
+
+The following lines look like ASCII graphics, but by now you should know
+enough to read them:
+
+```tla
+{{#include ../../../test/tla/bin-search/BinSearch4.tla:60:71}}
+```
+
+These lines are the indented form of `\/` for three cases:
+
+ - when `midVal < INPUT_KEY`, or
+ - when `midVal > INPUT_KEY`, or
+ - when `midVal = INPUT_KEY`.
+
+We could write these expressions with `IF-THEN-ELSE` or even with the TLA+
+operator `CASE` (see [Summary of TLA+][]). However, we find the disjunctive
+form to be the least cluttered, though unusual.
+
+Now we can check the postcondition again:
+
+```sh
+$ apalache-mc check --inv=Postcondition MC4_8.tla
+```
+
+The check goes through, but did it do much? Recall, that we fixed `INPUT_SEQ`
+to be the empty sequence `<< >>` in `MC4_8.tla`. Hence, we never enter the loop
+we have just specified.
+
+Actually, Apalache is giving us a hint that it has never tried some of the
+cases:
+
+```
+...
+PASS #13: BoundedChecker
+State 0: Checking 1 state invariants
+Step 0: picking a transition out of 1 transition(s)
+Step 1: Transition #0 is disabled
+Step 1: Transition #1 is disabled
+Step 1: Transition #2 is disabled
+Step 1: Transition #3 is disabled
+State 1: Checking 1 state invariants
+Step 1: picking a transition out of 1 transition(s)
+Step 2: Transition #0 is disabled
+Step 2: Transition #1 is disabled
+Step 2: Transition #2 is disabled
+Step 2: Transition #4 is disabled
+Step 2: picking a transition out of 1 transition(s)
+...
+```
+
+**Digression: symbolic transitions.** Internally, Apalache decomposes the
+predicates `Init` and `Next` into independent pieces like `Init == Init$0 \/
+Init$1` and `Next == Next$0 \/ Next$1 \/ Next$2 \/ Next$3`. If you want to see,
+how it is done, run Apalache with the options `--write-intermediate` and `--run-dir`:
+
+```sh
+$ apalache-mc check --inv=Postcondition --write-intermediate=1 --run-dir=out MC4_8.tla
+```
+
+Check the file `out/intermediate/09_OutTransition.tla`, which contains the
```

Comment:
As a general question, how much can we keep printed outputs and paths in sync with our implementation?

Reply:
If we change the sequence of passes in the code, then the number will change.